python/app/dialog.py

```diff
@@ -892,6 +903,43 @@ def _change_work_area(self, entity_type, entity_id):
 
                 self._do_work_area_switch(entity_type, entity_id)
 
+    def _update_preset_filters_on_tab_change(self, index):
+        """
+        Update the preset filters for the given tab when the tab is changed.
+        :param index: index of the tab
+        """
+        tab_name = self._current_entity_tabs[index]
+        tab = self._entity_tabs.get(tab_name, None)
+        if not tab:
+            return
+
+        filter_menu = tab.get("filter_menu", None)
+        if not filter_menu:
+            return
+
+        # This prevents any potential filter changes caused by the hook creating a non-deterministic result
+        # from triggering a data load
+        filter_menu.blockSignals(True)
+        self._update_preset_filters(tab_name, tab, filter_menu)
+        filter_menu.blockSignals(False)
```

Comment:
We can remove this function altogether (see comment above).

Reply:
As above happy to try something different, but need a way to keep the filters fresh. Changing tab seemed like a good time to refresh them.

Reply:
Ok, I'm ok to keep this as is then. Just a small change to this though to ensure Qt signals are restored and the state is restored to the last previous state:

```suggestion
    def _update_preset_filters_on_tab_change(self, index):
        """
        Update the preset filters for the given tab when the tab is changed.
        :param index: index of the tab
        """
        tab_name = self._current_entity_tabs[index]
        tab = self._entity_tabs.get(tab_name, None)
        if not tab:
            return

        filter_menu = tab.get("filter_menu", None)
        if not filter_menu:
            return

        # This prevents any potential filter changes caused by the hook creating a non-deterministic result
        # from triggering a data load
        restore_state = filter_menu.blockSignals(True)
        try:
            self._update_preset_filters(tab_name, tab, filter_menu)
        finally:
            filter_menu.blockSignals(restore_state)
```